### ISSUE TITLE
Updating alt text for images on Geo.ca

### ISSUE THREAD

**Issue by Andrea-Merry** [Canadian-Geospatial-Platform/geo.ca#65]
_Source: Email from Carli Rosenbaum on October 26th, 2022_

With the first few months of reporting, we have come across items to be updated to improve analytics tracking. The update is as follows:

**Update alt text for images**
When a visitor clicks on an image link, some of the data that is collected is the alt text. For example, when I click on the “GEO in the News” image, what is captured in some variables is the alt text for the image which is “Hand reaching towards sunset”. This could be problematic in the future especially if the link changes and the image alt text remains the same. Would it be possible to have the image alt values to be more descriptive i.e. “Geo in the News link: Hand reaching towards sunset image”

**Comment by jvanulde** [comment]
I believe that the links should be implemented as:

`<a href="#" alt="Geo in the News link"><img src="#" alt="Hand reaching towards sunset image" /></a>` 

@indiciumx do you have thoughts on this? @Andrea-Merry can you check with the originator to see if the above solution would work?

**Comment by Andrea-Merry** [comment]
@indiciumx I will check with Carli and get back to you.

**Comment by Andrea-Merry** [comment]
Source: _Email from Carli Rosenbaum on November 1, 2022_
@jvanulde @indiciumx

Unfortunately, the Adobe Analytics tracking is setup that when an a image is used as a “link”, the alt text for that image is collected as the click value. In the case for the proposed solution, the value that would be collected is “Hand reaching towards sunset image” rather than the alt text in the <a> tag (Geo in the News link). 

Here is a solution for how the tracking would collect a more descriptive value.
`<a href="#"><img src="#" alt="Geo in the News link: Hand reaching towards sunset image" /></a>`

Please let me know if there is anything else. 

Thank you!
Carli

**Comment by jvanulde** [comment]
The problem is that solution would fail accessibility guidelines. The link should have alt text. We'll likely need to duplicate the alt text. @indiciumx can estimate the level of effort to make this change?

**Comment by elatsis** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Canadian-Geospatial-Platform-62d04ca87c442300145c51fa/issues/Canadian-Geospatial-Platform/geo.ca/65?planning-poker) with Zenhub @jvanulde @indiciumx

**Comment by jaredkinger** [comment]
This seems like a bit of a non issue to me. If the website content changes the analytics tracking should reflect that change.

Alt text is primarily for screen readers and accessibility. Adding additional information to the alt text that is purely for analytical organization would be detrimental to those users.

From a maintenance standpoint this would create a lot of clutter and duplication. The alt text is saved to an image's metadata in WordPress so if we had an image that was used across multiple pages with different links on each of those pages we would need to upload a duplicate of that image for each different link to change the alt text.

Adding the alt attribute to an `<a>` tag would be invalid html so I wouldn't recommend that either. 

@Andrea-Merry Can we get some more context and information on what report they are running that this would be an issue for?

**Comment by jaredkinger** [comment]
@Andrea-Merry Can I get an update on this?

**Comment by Andrea-Merry** [comment]
@jaredkinger Hi. Sorry for the delay. Carli no longer works here. I sent her replacement Dawn Hoover an email this afternoon with the back and forth conversation and I asked her to see if this is still an issue.  Unfortunately, I think Dawn and her team is on strike. I made a note to follow up.

**Comment by Andrea-Merry** [comment]
@jaredkinger Dawn was on strike. Here is her response. Let me know if you need anything else. Thank you, Andrea.
Hi Andrea,
I’ve been in the web development space for two decades and this is my recommendation:
1.	If there is an image, hyperlinked or not, it should have an ALT description in the tag - <a href="#"><img src="#" alt="Geo in the News link: Hand reaching towards sunset image" /></a> Their primary use is for accessibility (screen readers), and secondly for search engine optimization.
a.	An example is the GoC logo at the top of every webpage: 
 
It’s an image and a link. The code is:
<a href=[https://www.canada.ca/en.html](https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.canada.ca%2Fen.html&data=05%7C01%7Candrea.merry%40NRCan-RNCan.gc.ca%7C8b2f7b042aa84222403208db4bfb902f%7C05c95b3390ca49d5b644288b930b912b%7C0%7C0%7C638187316049235689%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=je3OmkqnZL5OjUYcll6HLE1PcEXXKR40t3JtJsLIEuY%3D&reserved=0) property="url"> NO ALT <img src="/sites/all/libraries/canadaca/GCWeb/assets/sig-blk-en.svg" class="logo-en" alt="Government of Canada" property="logo"><span class="wb-inv"> / <span lang="fr">Gouvernement du Canada</span></span></a>

2.	If there is a text link on a page (not an image), alt text can be used to describe a link but the proper was is the link itself should be descriptive (absolutely no “click here” or “read more” text as the link) as per [WCAG 2.0 success criteria](https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.w3.org%2FTR%2FWCAG20-TECHS%2FF84.html&data=05%7C01%7Candrea.merry%40NRCan-RNCan.gc.ca%7C8b2f7b042aa84222403208db4bfb902f%7C05c95b3390ca49d5b644288b930b912b%7C0%7C0%7C638187316049235689%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=bnEqpKProbs3huIFzmVRCjKyJKIYp0Gu%2F5kGjIr1M7k%3D&reserved=0). 
3.	I have not yet come across a situation in Web Analytics where I’ve needed an alt tag to determine a stat metric. BUT, I have come across links that are repetitive throughout a webpage (ex. Contact us used repeatedly) and I have not been able to determine how many times each one was clicked because they are all the same and nothing is distinguishing them from one another. In these cases, I recommend the links use different text (descriptive text as per point 2. above) because we can determine the link text (name) metric. Ex. “contact us to apply for the program” and “contact us to learn more about the program” instead of both saying, “contact us”. 

The best thing to do when developing a website, is to test it in a screen reader and see if what the screen reader reads, makes sense. There is a free Google Chrome extension to hear what people experience with a screen reader: [https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn/related?hl=en](https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fchrome.google.com%2Fwebstore%2Fdetail%2Fscreen-reader%2Fkgejglhpjiefppelpmljglcjbhoiplfn%2Frelated%3Fhl%3Den&data=05%7C01%7Candrea.merry%40NRCan-RNCan.gc.ca%7C8b2f7b042aa84222403208db4bfb902f%7C05c95b3390ca49d5b644288b930b912b%7C0%7C0%7C638187316049235689%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=tgyaYdPpj4vRiQR4yJxePQoefjFG6VlBcDdupcYnPyA%3D&reserved=0). 

I hope this helps!
Dawn
